swarm manage panic error when docker-engine 1.10.x set a label which is not key=value format

docker-engine 1.10.x has a new feature of new configure json file `/etc/docker/daemon.json`, and we can set labels in this json file like below:
```
root@ubuntu:~# cat /etc/docker/daemon.json
{
	"labels":["a","b"]
}
```

`docker info` output:
```
root@ubuntu:~# docker info
Containers: 1
 Running: 0
 Paused: 0
 Stopped: 1
Images: 18
Server Version: 1.10.0
......
Labels:
 a
 b
``` 

If we use flags to set labels in docker daemon, daemon will validate lablel's format "key=value".However setting labels in daemon.json can escape validation,like above just key without value. It is a **BUG** of docker daemon.

And this case happens, as Swarm regards each label as format "key=value" by default, swarm run into a panic error, like below:
```
panic: runtime error: index out of range

goroutine 12 [running]:
github.com/docker/swarm/cluster.(*Engine).updateSpecs(0xc82011a6c0, 0x0, 0x0)
	/root/swarm/src/github.com/docker/swarm/cluster/engine.go:426 +0x8c9
github.com/docker/swarm/cluster.(*Engine).ConnectWithClient(0xc82011a6c0, 0x7fd39cc4d908, 0xc820108960, 0x0, 0x0)
	/root/swarm/src/github.com/docker/swarm/cluster/engine.go:188 +0x5b
github.com/docker/swarm/cluster.(*Engine).Connect(0xc82011a6c0, 0x0, 0x0, 0x0)
	/root/swarm/src/github.com/docker/swarm/cluster/engine.go:163 +0x259
github.com/docker/swarm/cluster/swarm.(*Cluster).validatePendingEngine(0xc82010a8c0, 0xc82011a6c0, 0xc82011a600)
	/root/swarm/src/github.com/docker/swarm/cluster/swarm/cluster.go:281 +0x5b
created by github.com/docker/swarm/cluster/swarm.(*Cluster).addEngine
	/root/swarm/src/github.com/docker/swarm/cluster/swarm/cluster.go:272 +0x354
```

and the code is very simple, and is just with no validation:
```
	for _, label := range info.Labels {
		kv := strings.SplitN(label, "=", 2)
		e.Labels[kv[0]] = kv[1]
	}
```

--- a/cluster/engine.go
+++ b/cluster/engine.go
@@ -423,6 +423,10 @@ func (e *Engine) updateSpecs() error {
 	}
 	for _, label := range info.Labels {
 		kv := strings.SplitN(label, "=", 2)
+		if len(kv) != 2 {
+			message := fmt.Sprintf("Engine (ID: %s, Addr: %s) contains an invalid label (%s) not formatted as \"key=value\".", e.ID, e.Addr, label)
+			return fmt.Errorf(message)
+		}
 		e.Labels[kv[0]] = kv[1]
 	}
 	return nil

